R/jira.R

```diff
@@ -563,3 +563,154 @@ create_status <- function(jira_domain_url, status) {
+#' messages or not
+#' @param maxDownloads Maximum downloads per function call
+#' @param created_latest the maximum value of the 'created' field in existing files
+#' This parameter is set to enable refresh capability
+#' @export
+download_and_save_jira_issues <- function(domain,
+                                          username = NULL,
+                                          password = NULL,
+                                          jql_query,
+                                          fields,
+                                          save_path_issue_tracker_issues,
+                                          maxResults = 50,
+                                          verbose = FALSE,
+                                          maxDownloads = 5000,
+                                          created_latest = "'1970-01-01'") {
+
+  # Ensure the domain starts with https:// for secure communication.
+  if (!grepl("^https?://", domain)) {
+    domain <- paste0("https://", domain)
+  }
+
+  if (!is.null(created_latest)){
+     jql_query <- paste(jql_query, "AND created >= ", created_latest)
+     message(jql_query)
+   }
+
+  #Initialize variables for pagination
+
+  startAt <- 0
+  total <- maxResults
+  all_issues <- list()
+  # This variable counts your download count. This is important to not exceed the max downloads per hour
+  downloadCount <- 0
+  time <- Sys.time()
+  message("Starting Downloads at ", time)
+  # Loop that downloads each issue into a file
+  repeat{
+
+    # Check update our download count and see if it is approaching the limit
+    if (downloadCount + maxResults > maxDownloads) {
+      # erorr message
+      time <- Sys.time()
+      message("Cannot download as maxDownloads will be exceeeded. Reccommend running again at a later time. Downloads ended at ", time)
+      break
+    } # Resume donwloading
+
+    # Construct the API endpoint URL
+    url <- paste0(domain, "/rest/api/2/search")
+
+    # Authenticate if username and password are provided
+    if(length(credentials) >= 2) {
+      username <- credentials[1]
+      password <- credentials[2]
+      # Use the credentials for authentication
+      auth <- httr::authenticate(as.character(username), as.character(password), "basic")
+    } else {
+      if(verbose){
+        message("No credentials present.")
+      }
+      auth <- NULL
+    }
+
+    # Prepare query parameters for the API call
+    query_params <- list(jql = jql_query, fields = paste(fields, collapse = ","), maxResults = maxResults, startAt = startAt)
+
+    # Make the API call
+    response <- httr::GET(url, query = query_params)
+
+    # Stop if there's an HTTP error
+    if (httr::http_error(response)) {
+      stop("API request failed: ", httr::http_status(response)$message)
+    }
+
+    # Extract issues. Append the new issues to all_issues
+    content <- jsonlite::fromJSON(httr::content(response, "text", encoding = "UTF-8"), simplifyVector = FALSE)
```

Comment:
Try using read_json here instead of fromJSON. I am also curious if you need to parse the JSON in memory just to save it. Can you just save the content you downloaded instead, without parsing? 

You can still parse to extract information needed, my question is whether saving the raw object itself rather than save the parsed one does not make more sense.

Reply:
It changes when you generate the docs!